engine/src/flutter/shell/platform/android/test/io/flutter/embedding/engine/FlutterShellArgsTest.java

```diff
@@ -2,35 +2,87 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package test.io.flutter.embedding.engine;
+package io.flutter.embedding.engine;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
-import android.content.Intent;
-import androidx.test.ext.junit.runners.AndroidJUnit4;
-import io.flutter.embedding.engine.FlutterShellArgs;
-import java.util.Arrays;
-import java.util.HashSet;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 import org.junit.Test;
-import org.junit.runner.RunWith;
 
-@RunWith(AndroidJUnit4.class)
 public class FlutterShellArgsTest {
+
   @Test
-  public void itProcessesShellFlags() {
-    // Setup the test.
-    Intent intent = new Intent();
-    intent.putExtra("dart-flags", "--observe --no-hot --no-pub");
-    intent.putExtra("trace-skia-allowlist", "skia.a,skia.b");
+  public void allFlags_containsAllFlags() {
```

Comment:
I know reflection is gross but I love reflection in tests that prevents me from making a mistake.